backend/src/facebook-feed/facebook-feed.service.ts

```diff
@@ -19,34 +21,40 @@ export class FacebookFeedService {
     }
 
     async fetchFacebookFeeds() {
-        try {
-            this.logger.log('Fetching facebook feeds Started');
-            const vantaaFacebookDataUrl = `${config.facebookGraphApi}${sources.vantaa}/posts?`
-                + `fields=${queryParams.join(',')}`
-                + `&access_token=${config.facebookPageToken}`;
-            const facebookFeeds = this.transformData((await axios.get(vantaaFacebookDataUrl)).data.data);
-            await this.feedService.saveFeeds(facebookFeeds);
-            this.logger.log('Fetching facebook feeds Completed');
-        } catch (error) {
-            this.logger.error(`Failed to get facebook feeds: ${error}`)
-        }
+        this.logger.log('Fetching facebook feeds Started');
+        FB.api(`${sources.vantaa}/posts`, { fields: queryParams })
+            .then(this.transformData)
+            .then(this.persistIntoDb)
+            .catch(error => this.logger.error(`Failed to get facebook feeds: ${error}`));
+        this.logger.log('Fetching facebook feeds Completed');
+    }
+
+    persistIntoDb = feeds => this.feedService.saveFeeds(feeds);
+
+    transformData = async (feeds) => {
+        return await Promise.all(
+            feeds.data.map(async feed => {
+                let profileImage = await this.getProfileImage(feed.from.id);
+                const feed_id = feed.id;
+                delete feed.id;
+                return {
+                    author: feed.from.name,
+                    author_thumbnail: profileImage,
+                    likes: feed.reactions ? feed.reactions.data.length : null,
+                    description: feed.story ? feed.story : feed.message,
+                    source: 'Facebook',
+                    title: feed.status_type,
+                    pub_date: feed.created_time,
+                    image_url: feed.picture,
+                    feed_id
+                }
+            }));
     }
 
-    transformData = feeds => {
-        return feeds.map(feed => {
-            const feed_id = feed.id;
-            delete feed.id;
-            return {
-                ...feed,
-                comments: feed.comments ? feed.comments.data : null,
-                likes: feed.reactions ? feed.reactions.data.length : null,
-                description: feed.story ? feed.story : feed.message,
-                source: 'Facebook',
-                title: feed.status_type,
-                pub_date: feed.created_time,
-                image_url: feed.picture,
-                feed_id
-            }
-        });
+    getProfileImage = (id: number) => {
```

Comment:
usually `get` functions are returning data synchronously
Like, when you have a class Car, and you `getName` it will return directly the name.

It seems here that you are trying to `fetch` more remote data instead, so I suggests the name `fetchProfileImage`.
This function should just return a Promise. Consumption from data delivered from this Promise, or Rejection handling should be specified by consumer (caller code)